io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -308,71 +325,101 @@ impl NvmfSubsystem {
         event: spdk_nvmf_subsystem_events,
     ) {
         let ss = NvmfSubsystem::from(subsystem);
+
         let subsys_nqn = unsafe {
             spdk_nvmf_subsystem_get_nqn(ss.0.as_ptr())
                 .as_str()
                 .to_string()
         };
-        debug!(
-            "nvmf_event_handler called {:?} for subsys {subsys_nqn}",
-            event
-        );
+
         let spdk_ctrlr =
             SpdkNvmfController::from(cb_arg as *mut spdk_nvmf_ctrlr);
 
-        debug!("host controler nqn {:?} ", spdk_ctrlr);
+        debug!(
+            "NVMF event handler: {event:?} for subsys '{subsys_nqn}'; \
+            host controler: {spdk_ctrlr:?}"
+        );
 
         let nexus_name = match extract_nexus_name(&subsys_nqn) {
             Some(value) => value,
             None => {
-                debug!("subsys {subsys_nqn} is not a valid nexus");
+                warn!(
+                    "NVMF event handler: subsys '{subsys_nqn}' is not a nexus"
+                );
                 return;
             }
         };
 
-        let mut nexus = match nexus_lookup_mut(&nexus_name) {
+        let nex = match nexus_lookup(&nexus_name) {
             Some(v) => v,
             None => {
-                debug!(
-                    "subsys {subsys_nqn} does not contain a valid nexus name"
+                warn!(
+                    "NVMF event handler:: subsys '{subsys_nqn}' does not \
+                    contain a valid nexus name"
                 );
                 return;
             }
         };
-        debug!("Nexus found for subsys {subsys_nqn}");
+
         let hostnqn = spdk_ctrlr.hostnqn();
         match event {
             SPDK_NVMF_SS_INIATOR_TIMEOUT => {
-                unsafe {
-                    nexus.as_mut().rm_initiator(&hostnqn);
+                info!(
+                    "NVMF event handler: {nex:?}: \
+                    initiator timed out: '{hostnqn}'"
+                );
+
+                nex.rm_initiator(&hostnqn);
+
+                if !ENABLE_NEXUS_RESET.load(Ordering::SeqCst) {
```

Comment:
This check needs to be better moved to the place where this event handler is registered via
`spdk_nvmf_subsystem_register_for_event()`, otherwise we'd keep calling add_initiator and rm_initiator for other events.

Reply:
As discussed, we are intending to disable selectively only the reset event part here.